pkg/resource/managed_reconciler.go

```diff
@@ -323,6 +296,36 @@ func (r *ManagedReconciler) Reconcile(req reconcile.Request) (reconcile.Result,
 		return reconcile.Result{RequeueAfter: r.shortWait}, errors.Wrap(r.client.Status().Update(ctx, managed), errUpdateManagedStatus)
 	}
 
+	if meta.WasDeleted(managed) {
+		// TODO: Reclaim Policy should be used between Claim and Managed. For Managed and External Resource,
+		// we need another field.
+		if observation.ResourceExists && managed.GetReclaimPolicy() == v1alpha1.ReclaimDelete {
+			managed.SetConditions(v1alpha1.Deleting())
```

Comment:
I didn't notice that separation of level. Makes sense to leave it to ExternalClient. I'll document the pattern doc so that people know _they_ manage these conditions.